poetry/utils/shell.py

```diff
@@ -58,9 +58,6 @@ def activate(self, env):  # type: (VirtualEnv) -> None
                 self._path, ["-i"], dimensions=(terminal.height, terminal.width)
             )
 
-        if not self._name == "bash":
```

Comment:
Since we know that it works for `zsh` we can change it to

```python
if self._name == "zsh":
```

Reply:
Updated - please resolve and merge if all looks good to you!